pkg/virt-controller/network/ipamclaims_test.go

```diff
@@ -0,0 +1,208 @@
+	})
+
+	BeforeEach(func() {
+		vmi = api.NewMinimalVMI(vmiName)
+		vmi.Namespace = namespace
+		vmi.Spec.Domain.Devices.Interfaces = []virtv1.Interface{
+			{Name: redNetworkLogicalName},
+			{Name: blueNetworkLogicalName},
+			{Name: "absent", State: virtv1.InterfaceStateAbsent},
+		}
+		vmi.UID = vmiUID
+		vmi.Spec.Networks = []virtv1.Network{
+			logicalSecondaryNetwork(redNetworkLogicalName, redNetworkNadName),
+			logicalSecondaryNetwork(blueNetworkLogicalName, blueNetworkNadName),
+			logicalSecondaryNetwork("absent", "absent-net"),
+		}
+	})
+
+	Context("With allowPersistentIPs enabled in the NADs", func() {
+		BeforeEach(func() {
+			persistentIPs := map[string]struct{}{redNetworkNadName: {}, blueNetworkNadName: {}}
+			Expect(createNADs(networkClient, namespace, vmi.Spec.Networks, persistentIPs)).To(Succeed())
+		})
+
+		It("should create the expected IPAMClaims", func() {
+			ownerRef := &v1.OwnerReference{
+				APIVersion:         "kubevirt.io/v1",
+				Kind:               "VirtualMachineInstance",
+				Name:               vmiName,
+				UID:                vmiUID,
+				Controller:         pointer.P(true),
+				BlockOwnerDeletion: pointer.P(true),
+			}
+			ipamClaimsManager := network.NewIPAMClaimsManager(virtClient)
+			Expect(ipamClaimsManager.CreateIPAMClaims(vmi.Namespace, vmi.Name, vmi.Spec.Domain.Devices.Interfaces, vmi.Spec.Networks, ownerRef)).To(Succeed())
+
+			ipamClaimsList, err := virtClient.IPAMClaimsClient().K8sV1alpha1().IPAMClaims(namespace).List(
+				context.Background(),
+				v1.ListOptions{},
+			)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ipamClaimsList.Items).To(HaveLen(2))
+			assertIPAMClaim(ipamClaimsList.Items[0], blueNetworkLogicalName, "pod16477688c0e")
+			assertIPAMClaim(ipamClaimsList.Items[1], redNetworkLogicalName, "podb1f51a511f1")
+		})
+	})
+})
+
+var _ = Describe("GetNetworkToIPAMClaimParams", func() {
+	var virtClient *kubecli.MockKubevirtClient
+	var networkClient *fakenetworkclient.Clientset
+	var networks []virtv1.Network
+
+	BeforeEach(func() {
+		ctrl := gomock.NewController(GinkgoT())
+		virtClient = kubecli.NewMockKubevirtClient(ctrl)
+
+		networkClient = fakenetworkclient.NewSimpleClientset()
+		virtClient.EXPECT().NetworkClient().Return(networkClient).AnyTimes()
+	})
+
+	BeforeEach(func() {
+		networks = []virtv1.Network{
+			logicalSecondaryNetwork(redNetworkLogicalName, redNetworkNadName),
+			logicalSecondaryNetwork(blueNetworkLogicalName, blueNetworkNadName),
+			{Name: "default_pod_network", NetworkSource: virtv1.NetworkSource{Pod: &virtv1.PodNetwork{}}},
+			{Name: "default_multus", NetworkSource: virtv1.NetworkSource{Multus: &virtv1.MultusNetwork{NetworkName: "default_multus_net", Default: true}}},
+		}
+
+		persistentIPs := map[string]struct{}{redNetworkNadName: {}}
+		Expect(createNADs(networkClient, namespace, networks, persistentIPs)).To(Succeed())
+	})
+
+	It("should return the expected IPAMClaim parameters", func() {
```

Comment:
Please add two tests:
1. IPAMClaim already exist with the same owner ref.
2. IPAMClaim already exist with different owner ref.